### PR TITLE
Generic polling   wildfire

### PR DESCRIPTION
**Pull request by karinafishman**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-8981
relates: https://jira-dc.paloaltonetworks.com/browse/CIAC-8740

## Description
Deprecate for the old playbook of Wildfire

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/Detonate_URL_-_WildFire_v2.2.yml
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/Detonate_URL_-_WildFire_v2.2.yml:309]:
```suggestion
  description: File type, e.g., "PE".
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/Detonate_URL_-_WildFire_v2.2.yml
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/Detonate_URL_-_WildFire_v2.2.yml:327]:
```suggestion
  description: The indicator tested.
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/Detonate_URL_-_WildFire_v2.2.yml
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/Detonate_URL_-_WildFire_v2.2.yml:339]:
```suggestion
  description: The file object.
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/Detonate_URL_-_WildFire_v2.2_README.md
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/Detonate_URL_-_WildFire_v2.2_README.md:15]:
```suggestion
WildFire-v2
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/Detonate_URL_-_WildFire_v2.2_README.md
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/Detonate_URL_-_WildFire_v2.2_README.md:32]:
```suggestion
| URL | URL of the webpage or file URL to detonate. The URL is taken from the context. | URL.Data | Optional |
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/Detonate_URL_-_WildFire_v2.2_README.md
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/Detonate_URL_-_WildFire_v2.2_README.md:54]:
```suggestion
| DBotScore.Indicator | The indicator tested. | string |
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/Detonate_URL_-_WildFire_v2.2_README.md
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/Detonate_URL_-_WildFire_v2.2_README.md:58]:
```suggestion
| File | The file object. | unknown |
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2.yml
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2.yml:121]:
```suggestion
      description: Check if the integration instance is enabled.
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2.yml
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2.yml:191]:
```suggestion
      description: Check if the file type is supported.
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2.yml
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2.yml:228]:
```suggestion
        https://docs-cortex.paloaltonetworks.com/r/Cortex-XSOAR/6.12/Cortex-XSOAR-Administrator-Guide/Automations
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2.yml
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2.yml:284]:
```suggestion
        https://docs-cortex.paloaltonetworks.com/r/Cortex-XSOAR/6.12/Cortex-XSOAR-Administrator-Guide/Automations
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2.yml
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2.yml:344]:
```suggestion
      description: Check if there are any supported files to upload to WildFire for detonation.
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2.yml
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2.yml:455]:
```suggestion
  description: The duration after which to stop polling and to resume the playbook (in minutes).
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2.yml
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2.yml:470]:
```suggestion
  description: The file type, e.g., "PE".
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2_README.md
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2_README.md:23]:
```suggestion
SetAndHandleEmpty
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2_README.md
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2_README.md:38]:
```suggestion
| Timeout | The duration after which to stop polling and to resume the playbook \(in minutes\). | 15 | Optional |
```

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2_README.md
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/Playbooks/WildFire_-_Detonate_file_v2_README.md:49]:
```suggestion
| File.Type | The file type, e.g., "PE". | string |
```

File: Packs/Palo_Alto_Networks_WildFire/ReleaseNotes/2_1_37.md
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/ReleaseNotes/2_1_37.md:5]:
```suggestion
Deprecated. Use Detonate File From URL - WildFire v2 instead.
```

File: Packs/Palo_Alto_Networks_WildFire/ReleaseNotes/2_1_37.md
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/ReleaseNotes/2_1_37.md:9]:
```suggestion
New: Detonate one or more files using the Wildfire integration. This playbook returns relevant reports to the War Room and file reputations to the context data.
```

File: Packs/Palo_Alto_Networks_WildFire/ReleaseNotes/2_1_37.md
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/ReleaseNotes/2_1_37.md:15]:
```suggestion
Deprecated. Use Detonate URL - WildFire v2.2 instead.
```

File: Packs/Palo_Alto_Networks_WildFire/ReleaseNotes/2_1_37.md
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/ReleaseNotes/2_1_37.md:19]:
```suggestion
New: Detonate a webpage or remote file using the WildFire integration. This playbook returns relevant reports to the War Room and file reputations to the context data.
```

File: Packs/Palo_Alto_Networks_WildFire/ReleaseNotes/2_1_37.md
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/ReleaseNotes/2_1_37.md:25]:
```suggestion
Deprecated. Use Detonate URL - WildFire v2.2 instead.
```

File: Packs/Palo_Alto_Networks_WildFire/ReleaseNotes/2_1_37.md
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/ReleaseNotes/2_1_37.md:29]:
```suggestion
Deprecated. Use WildFire - Detonate File v2 instead.
```

File: Packs/Palo_Alto_Networks_WildFire/ReleaseNotes/2_1_37.md
Comment (ShirleyDenkberg) [inline Packs/Palo_Alto_Networks_WildFire/ReleaseNotes/2_1_37.md:33]:
```suggestion
New: Detonate one or more files using the Wildfire integration. This playbook
```

**Review by idovandijk — CHANGES_REQUESTED** [review]
Great job with the PR! Here are my comments, numbered so you can easily refer to them if you have questions:

### Detonate File From URL - Wildfire v2

1. Note how the condition checks if Wildfire **OR** Wildfire v2 is enabled. However, the command you're then executed is tied to Wildfire v2 only:
image

Since the old Wildfire version is deprecated, and we're releasing a new playbook, let's also fix the condition to check only if v2 is enabled.
Also, please note that as a best practice, we prefer to not tie the command to the integration brand unless we have a reason to
image

but since we really intend to only use Wildfire v2 here, we're OK here.

2. As we are tackling resource hogs, let's rethink the timeout. I think 15 is excessive. Try it on a medium size and larger size file please, and check the duration of the task (you will see the duration in the task itself), and let's tweak the timeout input so that it's at a safe level, yet not too long:
image

3. It seems that the "hash" argument is deprecated ;) let's use the SHA256 argument (and test it).

4. Please also update the playbook description (and generate new readme) to indicate that we're using Wildfire v2 (not Wildfire)

5. Please add the Unique transformer to the FileURL input, to ensure we don't accidentally process duplicate URLs (also part of our performance efforts):
image

6. We have new outputs from the `wildfire-report` command. Let's add those that we think are useful to the playbook outputs. Please make sure to output the sub-keys as well. For example, not only output the `WildFire.Report.ExtractedURL` object, but also the `WildFire.Report.ExtractedURL.URL` and `WildFire.Report.ExtractedURL.Verdict` outputs.
FYI - outputting the main key will also output the sub-keys. However, not outputting the sub-keys will cause the parent playbooks to be blind to them, so it's good practice to declare the sub-keys as well.

7. Looks like there are 2 tests for our playbook. Let's check if one of them is irrelevant - E.g., using the GenericPolling playbook, thus not really testing what we want.

### Detonate URL - Wildfire v2.2
Excellent job!
8. Let's tweak the unrealistic `Timeout` playbook input of 60 minutes to something much shorter.
9. Let's also add a Unique transformer to the `URL.Data` input
10. Please also update the playbook description to Wildfire v2, as well as the playbook README.

### Wildfire - Detonate File v2
11. Please add a Unique transformer to the File input.
12. Let's tweak the Timeout input here too.
13. I believe this condition should be changed to "Is Not Empty". The reason is that in the previous tasks we set those keys, and if there's no value, they'll be set with an empty value. This will cause the `Is Defined` condition to pass through because the key is defined, but it's empty. I'm not entirely sure if my theory is correct here, but it's worth being on the safe side.
14. In the Get Report task, let's use the non deprecated argument SHA256 (and test):
image
15. Please add more outputs that you think could be useful, to the playbook outputs.
16. Please update the playbook description to be Wildfire v2 (and the playbook README).
17. Please check if we can remove a test from the "tests" section of our playbook YML file. If one of the tests is not testing the file detonation flow, it's irrelevant.

### Deprecated Playbooks
18. Please update the README of the deprecated playbooks, so that it will be documented on our documentation portal too.

### Release Notes
19. While the release-notes are correct, it's worth adding a line explaining that the v2/v2.2 playbook features polling at the integration level, instead of using the GenericPolling playbook.

Let me know if you're unsure about anything or need help! Awesome job.

**Review by idovandijk — COMMENTED** [review]
Looks good, however I see that point 11 is still missing (Unique for the File playbook input), and see that point 1 is still as before (I remember you wanted to change it - your call).
Once these are fixed let me know and I'll approve

**Review by idovandijk — APPROVED** [review]
Well done

**Pull request merged**
